File: 045. Jump Game II/Java/Jump Game II.java
public class Solution {
    public int jump(int[] nums) {
        int result = 0;
        int index = 0;
        int current = 0;
        int next = 0;
        while (current < nums.length - 1) {
            result++;
            while (index <= current) {
                next = Math.max(next, index + nums[index]);
                index++;
            }
            current = next;
        }
        return result;
    }
}

File: 045. Jump Game II/Python/Jump Game II.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        index = 0
        current = 0
        next = 0
        while current < len(nums) - 1:
            result += 1
            while index <= current:
                next = max(next, index + nums[index])
                index += 1
            current = next
        return result
